backend/server/src/main/java/ai/verta/modeldb/authservice/MDBRoleServiceUtils.java

```diff
@@ -318,14 +319,10 @@ public String buildReadOnlyRoleBindingName(
   @Override
   public Workspace getWorkspaceByWorkspaceName(
       UserInfo currentLoginUserInfo, String workspaceName) {
-    /*from the name for workspace, get the workspace id and type.
-    if no workspace is present assume user's personal workspace*/
-    if (workspaceName == null || workspaceName.isEmpty()) {
-      return authService.workspaceIdByName(
-          true, authService.getUsernameFromUserInfo(currentLoginUserInfo));
-    } else {
-      return authService.workspaceIdByName(true, workspaceName);
+    if (Strings.isNullOrEmpty(workspaceName)) {
+      return Workspace.newBuilder().build();
```

Comment:
Have you checked what happens if we are not returning the workspace? I think we have to call UAC to get a workspace for the login user. might be the default workspace?

Reply:
The getWorkspaceByWorkspaceName returns a workspace which is used in setResource method. But if we use empty workspace when calling setResource then setResource by itself finds default workspace for the current user. So I think there is no need to search for default workspace during getWorkspaceByWorkspaceName method